hooks/useEns.ts

```diff
@@ -0,0 +1,72 @@
+import { useMemo, useEffect, useState } from 'react';
+import { utils, providers } from 'ethers';
+
+import Connector from 'containers/Connector';
+
+import { NetworkId } from 'constants/networks';
+
+import { hasNonAsciiCharacters, isEns } from 'utils/string';
+
+export const useEnsToAddress = (ensName: string) => {
+	const { network } = Connector.useContainer();
+	const [address, setAddress] = useState(ensName);
+	const provider = useMemo(
+		() =>
+			new providers.InfuraProvider(NetworkId.Mainnet, process.env.NEXT_PUBLIC_INFURA_PROJECT_ID),
+		[]
+	);
+
+	useEffect(() => {
+		const getAddress = async () => {
+			if (!isEns(ensName)) return ensName;
+			const address = await provider?.resolveName(ensName);
+			setAddress(address || ensName);
+		};
+
+		getAddress();
+	}, [ensName, provider, network.id]);
+
+	return address;
+};
+
+export const useAddressesToEns = (addresses: string[]) => {
+	const [ensNames, setEnsNames] = useState(addresses);
+	const { network } = Connector.useContainer();
+
+	const provider = useMemo(
+		() =>
+			new providers.InfuraProvider(NetworkId.Mainnet, process.env.NEXT_PUBLIC_INFURA_PROJECT_ID),
+		[]
+	);
+
+	useEffect(() => {
+		const getEnsName = async () => {
+			const areAddresses = addresses.map(utils.isAddress).some(Boolean);
```

Comment:
Shouldn't it be `.all(Boolean)`. What happens if one element in the array isn't an address?